source/resource/demo/templates/ChartSourcePlugin.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * TODO: Describe your plugin here
+ *
+ * @author add your name here
+ * @since ###SINCE###
+ */
+class $0ChartSource {
+  _chart = null;
+  _config = null;
+
+  /**
+   *
+   * @param config {object} the parsed src value of the dataset
+   * @param chart {cv.ui.structure.tile.components.Chart} instance of the chart that uses this plugin
+   */
+  constructor(config, chart) {
+    /*
+    <dataset src="plugin+mychartsource://authority@name/path?key=value" /> will be parsed to a config like this:
+    {
+       type: 'plugin,
+       subType: 'mychartsource,
+       authority: 'authority',
+       name: 'name',
+       path: '/path',
+       params: {
+          key: 'value',
+      }
+    */
+    this._config = config;
+    this._chart = chart;
+  }
+
+  /**
+   * Generate the request configuration for the chart component or disable it if the plugin
+   * fetches the data itself.
+   *
+   * If this plugin fetches the data itself, it should return {fetch: false} to disable the charts builtin data fetching.
+   * And it must implement the fetchData method.
+   *
+   * @param start {number} start time as unix timestamp
+   * @param end {number} end time as unix timestamp
+   * @param series {string} series name e.g. 'day', 'month', 'year'
+   * @param offset {number} can be used as alternative to calculate the start together with series, e.g. series 'day' and offset 1 -> start = end - 1 day
+   * @returns {
+   *    url: string,
+   *    options: object,
+   *    proxy: boolean
+   * } | { fetch: boolean }
+   */
+  getRequestConfig(start, end, series, offset) {
+    // return either a configuration to fetch the data from or {fetch: false} if the plugin fetches the data itself
+    // you can look at CometVisus builtin chart sources to see working examples: https://github.com/CometVisu/CometVisu/tree/develop/source/class/cv/io/timeseries
+  }
+
+  /**
+   * If the plugin request the data from an external source this function is called.
+   * getRequestConfig() must return {fetch: false} to disable the charts builtin data fetching and
+   * call this function instead.
+   * @param start {number} start time as unix timestamp
+   * @param end {number} end time as unix timestamp
+   * @param series {string} series name e.g. 'day', 'month', 'year'
+   * @param offset {number} series offset, can be used together with series to calculate the start date e.g. series: day, offset: 1 -> start = end - 1 day
+   * @returns {Promise<Array<[timestamp: number, value: number]>>}
```

Comment:
Shouldn't that be `{timestamp:...}`?

Reply:
No its an array of size 2: [timestamp, value].